### PR TITLE
Allow creating a duel-type bet

### PR DESCRIPTION
**Pull request by SimBuckeye**

### CODE REVIEW

File: src/app/betting/feature/place-bet-page.component.ts [hunk 1]
```diff
@@ -66,13 +69,55 @@ import { BetStatus } from '../../shared/util/supabase-helpers';
       />
     </label>
 
-    <!-- Bet terms -->
+    <!-- Bet Type Dropdown -->
+    <!-- eslint-disable-next-line -->
```
Comment (c-o-l-i-n) [inline src/app/betting/feature/place-bet-page.component.ts]: See comment in #17

File: src/app/betting/feature/place-bet-page.component.ts [hunk 2]
```diff
@@ -66,13 +69,55 @@ import { BetStatus } from '../../shared/util/supabase-helpers';
...
-      Bet Terms
-      <textarea rows="2" pInputTextarea [(ngModel)]="terms" [required]="true">
-      </textarea>
+      Bet Type
+      <p-dropdown
+        [options]="betTypes"
+        [(ngModel)]="selectedBetType"
+        optionLabel="betTypeString"
+        optionValue="betType"
+        styleClass="flex"
```
Comment (c-o-l-i-n) [inline src/app/betting/feature/place-bet-page.component.ts]: See comment in #17

File: src/app/betting/feature/place-bet-page.component.ts [hunk 3]
```diff
@@ -66,13 +69,55 @@ import { BetStatus } from '../../shared/util/supabase-helpers';
...
+      />
     </label>
 
+    @if (selectedBetType() === BetType.DuelWinner) {
```
Comment (c-o-l-i-n) [inline src/app/betting/feature/place-bet-page.component.ts]: See comment in #17

File: src/app/betting/feature/place-bet-page.component.ts [hunk 4]
```diff
@@ -219,8 +361,51 @@ export default class PlaceBetPageComponent {
...
+  generateBetTypeObject(type: BetType) {
+    switch (type) {
+      case BetType.DuelWinner:
+        return { betType: type, betTypeString: 'Duel Winner' };
+      default:
+        return { betType: BetType.Manual, betTypeString: 'Manual' };
+    }
+  }
+
+  private generateDetails(): Json {
```
Comment (c-o-l-i-n) [inline src/app/betting/feature/place-bet-page.component.ts:374]: See comment in #17

**Review by c-o-l-i-n — CHANGES_REQUESTED** [review]
I probably should have reviewed this PR before #17. All my comments are the same lol

File: src/app/betting/feature/place-bet-page.component.ts (hunk 4 above)
Comment (SimBuckeye) [reply src/app/betting/feature/place-bet-page.component.ts:374]: I'll plan on fixing this on the other branch so that I don't have to deal with a merge

**Review by c-o-l-i-n — APPROVED** [review]

**Pull request merged**